client/plots/profilePlot.js

```diff
@@ -14,92 +14,73 @@ export class profilePlot {
-		this.incomeSelect.selectAll('option').property('selected', d => d == this.income)
-		if (this.facilitySelect)
-			this.facilitySelect.selectAll('option').property('selected', d => d == this.config.facility)
-
-		if (this.selectComp) this.selectComp.selectAll('option').property('selected', (d, i) => i == this.componentIndex)
-	}
-
 	async init(appState) {
-		const holder = this.opts.holder.append('div')
+		const controlsDiv = this.opts.holder.append('div').style('display', 'inline-block')
+		const holder = this.opts.holder.append('div').style('display', 'inline-block').style('display', 'inline-block')
+
 		const div = holder.append('div').style('margin-left', '50px').style('margin-top', '20px')
 		const firstDiv = div.append('div').style('display', 'inline-block')
 		const plotDiv = holder.append('div')
 		this.dom = {
+			controlsDiv,
 			holder,
 			firstDiv,
 			filterDiv: div,
 			facilityDiv: div.insert('div').style('display', 'inline-block'),
 			plotDiv
 		}
 		const config = appState.plots.find(p => p.id === this.id)
-
 		this.sampleidmap = await this.app.vocabApi.getAllSamplesByName()
 		this.regions = [
-			{ key: '', label: '' },
-			{ key: 'Global', label: 'Global' }
+			{ value: '', label: '' },
+			{ value: 'Global', label: 'Global' }
 		]
-		this.incomes = ['']
-		this.incomes.push(...config.incomes)
+		this.incomes = [{ value: '', label: '' }]
+		this.incomes.push(
+			...config.incomes.map(elem => {
+				return { label: elem, value: elem }
+			})
+		)
 
 		for (const region of config.regions) {
-			this.regions.push({ key: region.name, label: region.name })
-			for (const country of region.countries) this.regions.push({ key: country, label: `-- ${country}` })
+			this.regions.push({ value: region.name, label: region.name })
+			for (const country of region.countries) this.regions.push({ value: country, label: `-- ${country}` })
 		}
+	}
 
-		div.append('label').style('margin-left', '15px').html('Region:').style('font-weight', 'bold')
-		this.regionSelect = div.append('select').style('margin-left', '5px')
-		this.regionSelect
-			.selectAll('option')
-			.data(this.regions)
-			.enter()
-			.append('option')
-			.attr('value', d => d.key)
-			.html((d, i) => d.label)
-
-		this.regionSelect.on('change', () => {
-			const config = this.config
-			config.facility = ''
-			config.region = this.regionSelect.node().value
-			config.income = ''
-			const sampleId = parseInt(this.sampleidmap[config.region])
-			config.sampleName = config.region
-			config.filter = getSampleFilter(sampleId)
-			this.app.dispatch({ type: 'plot_edit', id: this.id, config })
-		})
-
-		div.append('label').style('margin-left', '15px').html(' or Income Group:').style('font-weight', 'bold')
-		this.incomeSelect = div.append('select').style('margin-left', '5px')
-		this.incomeSelect
-			.selectAll('option')
-			.data(this.incomes)
-			.enter()
-			.append('option')
-			.html((d, i) => d)
+	setRegion(region) {
```

Comment:
@airenzp Usually in mass charts, a config value only affects one input control or at most, it may affect the visibility of another input but not its value. Here, selecting a region value affect also the facility and income input values (by emptying those values), that doesn't seem intuitive. These changes are okay as they don't seem to affect other mass charts/plots, so it's not necessary to redo the logic so one config value only affects one input value and vice-versa.

Reply:
The filters logic will evolve later on to combine filters. Now they are exclusive and it is true, it is not intuitive. We actually had labels showing they were exclusive before moving it to the controls. Anyways I guess is ok to be able to provide an alternative callback in case you need to do more than just updating one config value.